clustermesh-apiserver/kvstoremesh/root.go

```diff
@@ -44,3 +59,85 @@ func NewCmd(h *hive.Hive) *cobra.Command {
 
 	return rootCmd
 }
+
+type params struct {
+	cell.In
+
+	// Client is the client targeting the local cluster
+	Client kvstore.Client
+
+	Metrics    kvstoremesh.Metrics
+	Shutdowner hive.Shutdowner
+	Log        *slog.Logger
+	SyncState  syncstate.SyncState
+}
+
+func registerLeaderElectionHooks(lc cell.Lifecycle, llc *LeaderLifecycle, params params) {
+	var wg sync.WaitGroup
+	ctx, cancel := context.WithCancel(context.Background())
+
+	lc.Append(cell.Hook{
+		OnStart: func(cell.HookContext) error {
+			wg.Add(1)
+			go func() {
+				runLeaderElection(ctx, params.Log, llc, params.Client, params.Shutdowner, params.Metrics, params.SyncState)
+				wg.Done()
+			}()
+			return nil
+		},
+		OnStop: func(hctx cell.HookContext) error {
+			if err := llc.Stop(params.Log, hctx); err != nil {
+				return err
+			}
+			cancel()
+			wg.Wait()
+			return nil
+		},
+	})
+}
+
+func runLeaderElection(ctx context.Context, log *slog.Logger, lc *LeaderLifecycle, client kvstore.Client, shutdowner hive.Shutdowner, kvMetrics kvstoremesh.Metrics, SyncState syncstate.SyncState) {
```

Comment:
Nit. I'd suggest simply passing the params struct, to reduce the number of parameters.